client/cli/src/lib.rs

```diff
@@ -316,18 +316,24 @@ pub fn init_logger(
 		"%Y-%m-%d %H:%M:%S%.3f".to_string()
 	});
 
-	let subscriber = FmtSubscriber::builder()
+	let subscriber_builder = FmtSubscriber::builder()
 		.with_env_filter(env_filter)
-		.with_writer(std::io::stderr)
+		.with_writer(std::io::stderr as _)
 		.event_format(logging::EventFormat {
 			timer,
 			ansi: enable_color,
 			display_target: !simple,
 			display_level: !simple,
 			display_thread_name: !simple,
 		})
-		.finish().with(logging::NodeNameLayer);
-
+		// TODO: Q - There's a small cost to this, do we make it opt-in/out with cli flag?
+		.with_filter_reloading();
```

Comment:
I investigated this and I think we should enable this by default until it can be proven to be significant for our use-case. In an artificial context the performance impact is significant: it's twice as slow, but we're talking about ~35ns –> ~70ns for the worst case. Once you add in all the filtering and other options we use the impact is negligible.

Benchmarks for a "complex" setup very similar to our own (but using a noop writer):

```
reload/complex-collector                                                                             
                        time:   [1.8996 us 1.9034 us 1.9076 us]
                        change: [-3.5626% -3.1266% -2.7425%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild
reload/complex-collector-reloadable                                                                             
                        time:   [1.9313 us 1.9382 us 1.9456 us]
                        change: [-1.9828% -1.7096% -1.4540%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild
```

Reply:
OK - nice to have some figures for that. In that case I'll just add an opt-out CLI option `--disable-log-reloading` or similar

Reply:
One reason for leaving it on by default is that making it optional is nasty; the reloading changes the types such that I think we'd have to duplicate a lot of code (essentially everything from here to the end of the method). I don't think it's worth it.

Reply:
OK, that's a good point. But if it's not too ugly then IMO I think it's worth having - simply because if it's an optional performance configuration option implemented in `tracing` then I think we should mirror it.

Reply:
I'll try it out and we can see in review

Reply:
I just added a commit to do this - it worked out OK, only requiring 2 lines of duplicated code - one to finish the builder, another to call a new function that accepts a generic Subscriber. I don't think there is any downside to having this feature.